### DISCUSSION TITLE
Error when using Particle Dust

### DISCUSSION THREAD

**Discussion by KyllianGamer** [Minestom/Minestom#721] [Help] [answered]
When I send a Particle packet with following code:
`ParticlePacket particle = ParticleCreator.createParticlePacket(Particle.DUST, player.getPosition().x(), player.getPosition().y(), player.getPosition().z(), 0f, 0f, 0f, 1);
player.sendPacket(particle);`

The client throws following error:
`Internal Exception: io.netty.handler.codec.DecoderExcepion: java.lang.IndexOutOfBoundsException: readerIndex(50) + length(4) exceeds writerIndex(50): PooledUnsafeDirectByteBuf(ridx: 50, widx: 50, cap: 50)`

(Other Particle types work)

**Comment by iam4722202468** [comment, marked as answer]
Could be caused by Dust particles requiring addition information.

```
public static ParticlePacket createParticle(double x, double y, double z, float r, float g, float b) {
        ByteBuffer bb = ByteBuffer.wrap(new byte[16]);
        bb.putFloat(r);
        bb.putFloat(g);
        bb.putFloat(b);
        bb.putFloat(1.0f);

        return new ParticlePacket(
                Particle.DUST.id(),
                true,
                x,
                y,
                z,
                0,
                0,
                0,
                0,
                1,
                bb.array()
        );
    }
```
    
    
    
    
Try this


You can check the particles here to see what data is needed for them https://wiki.vg/Protocol#Particle
